src/main/java/seedu/address/model/contact/Contact.java

```diff
@@ -65,8 +65,8 @@ public boolean isSameContact(Contact otherPerson) {
 
         return otherPerson != null
                 && otherPerson.getName().equals(getName())
-                && otherPerson.getEmail().equals(getEmail())
-                && otherPerson.getTelegramUsername().equals(getTelegramUsername());
+                && (otherPerson.getEmail().equals(getEmail())
+                || otherPerson.getTelegramUsername().equals(getTelegramUsername()));
```

Comment:
I think the correct implementation is to use OR? Because both email and telegram ID are unique. So, comparing just one of them should be sufficient?

For example, Person A and Person B have the same name and email but different telegram ID. If we impose that every field must be the same, then we will consider Person A and Person B to be different. This means Person A and Person B are physically 2 different person that share one email account which I think we won't see this kind of case in real life?

By the way, the above is the case if we treat Contact as real person. I would like to clarify if we define Contact as real person though? Or maybe it is okay to have 2 same person but different email because they have different roles? e.g.

Person A
Name : Halo
Email : Personal email
Telegram : @halo

Person B
Name : Halo
Email : Work email
Telegram : -

Though, in this case, I think the name of the contact should have been the one that are differentiated instead, to avoid confusion 😆

Reply:
I think Michael makes sense, but it should not be applied under the equals method. It sounds like a new method should be created for isSamePerson(). Matthias makes sense since the two objects should be exactly the same under the equals method.

Reply:
This PR was actually made before the decision for Telegram to be an optional field was made. So line 69 will be removed and we will use only name and email to ascertain if 2 contacts are the same.

Reply:
@murtubak Yes, if it's the `equals()` method than we probably should check all the field, but the method that we are discussing now is not the `equals()` method, it is the `isSameContact()` method.